src/TileEngine/Layer.cpp

```diff
@@ -69,7 +71,38 @@ Layer::Layer(const TiXmlElement* layerData, const geometry::Rectangle& bounds) :
       {
          std::string entry;
          std::getline(layerStream, entry, ',');
-         m_tileMap(x, y + m_heightOffset) = atoi(entry.c_str()) - 1;
+         m_tileMap(x, y + m_heightOffset) = std::stoi(entry.c_str()) - 1;
+      }
+   }
+}
+
+void Layer::forEachCollisionRect(std::function<void(const geometry::Rectangle&)>&& func) const
+{
+   if(m_heightOffset != 0)
```

Comment:
This ends up being a problem, because forest canopies are not at ground level but should imply impassibility. That means we should remove this early return. The only concern is that objects meant to be suspended in the air may affect collisions. So far, though, that's not something that has happened.